templates/definition/meter/sonnenbatterie_eco56.yaml

```
@@ -48,5 +51,37 @@ render: |
     source: http
     uri: http://{{ .host }}:{{ .port }}/rest/devices/battery
     jq: .M30 # SOC relative to usable capacity (.M05 # display SOC)
+  batterymode:
+    source: switch
+    switch:
+      - case: 1 # normal
+        set:
+          source: http
+          method: PUT
+          uri: http://{{ .host }}:{{ .port }}/rest/devices/battery/C06
+          body: '10' # Automatic
+      - case: 2 # hold
+        set:
+          source: http
+          method: PUT
+          uri: http://{{ .host }}:{{ .port }}/rest/devices/battery/C06
+          body: '20' # Standby
+      - case: 3 # charge
+        set:
+          source: sequence
+          set:
+            - source: http
+              method: PUT
+              uri: http://{{ .host }}:{{ .port }}/rest/devices/battery/C06
+              body: '55' # slave mode
+            - source: sleep
+              duration: 1s
+            - source: watchdog
+              timeout: 30s # 3 minutes without setting a value will stop all charging
+              set:
+                source: http
+                method: PUT
+                uri: http://{{ .host }}:{{ .port }}/rest/devices/battery/C24
+                body: '{{ .chargepower }}'
```

The current implementation only makes sense in a sequence, we could also change it to be standalone:
```suggestion
            - source: sleep
              duration: 1s
              set:
                source: watchdog
                timeout: 30s # 3 minutes without setting a value will stop all charging
                set:
                  source: http
                  method: PUT
                  uri: http://{{ .host }}:{{ .port }}/rest/devices/battery/C24
                  body: '{{ .chargepower }}'
```

---

I wouldn’t mix sleep and watchdog

---

What do you mean by mixing them? My comment was aimed at the fact that my current sleep implementation is only really useful if it is part of a sequence.
In this other form it could also be used standalone.